### PR TITLE
BF silent install

### PR DESCRIPTION
**Pull request by yarikoptic**
upon playing with merged #631 found that it started to spit out errors while trying to clone from http and testing without `/.git`.  so this one brings swallowing back and also enhances it to log loglevel so we could quickly test if e.g. any ERROR was logged

please review/improve/merge @datalad/developers 
ah nah -- will merge -- please whine @datalad/developers  ;)

### CODE REVIEW

File: datalad/utils.py [hunk 1]
```diff
@@ -668,20 +673,63 @@ def handle(self):
...
+                match = '[%s] ' % level if level else ''
+
+            if msg:
+                match += msg
+
+            if match:
+                (assert_re_in if regex else assert_in)(match, self.out, **kwargs)
+            else:
+                assert not kwargs, "no kwargs to be passed anywhere"
+                assert self.out, "Nothing was logged!?"
```
Comment (yarikoptic) [inline datalad/utils.py]: strange that this block is not covered :-/  oh well -- at least tests pass ;)

File: datalad/distribution/install.py [hunk 2]
```diff
@@ -538,6 +539,8 @@ def __call__(
 
             # We possibly need to consider /.git URL
             candidate_source_urls = [source_url]
+            # TODO: isn't this a duplicate of above logic/implementation
+            # in _install_subds_from_flexible_source????
```
Comment (bpoldrack) [inline datalad/distribution/install.py]: Yes, it is. There is some RF'ing to be done regarding `_install_subds_from_flexible_source`, `GitRepo.clone()` and this one. They all try to clone from possibly multiple URLs.

File: datalad/utils.py [hunk 3]
```diff
@@ -668,20 +673,63 @@ def handle(self):
...
+            # store for access while object exists
+            self._final_out = self.out
             self._out.close()
             out_name = self._out.name
             del self._out
             gc.collect()
             if not file_:
                 rmtemp(out_name)
 
+        def assert_logged(self, msg=None, level=None, regex=True, **kwargs):
```
Comment (bpoldrack) [inline datalad/utils.py]: Nice!

**Comment by bpoldrack** [comment]
Well, apparently we have new merge policy. ;-)

**Pull request merged**